test/Test_Elaboration.re

```diff
@@ -100,124 +115,314 @@ let u8: Exp.t =
+
+let undefined_str = "
+undef
+";
+let undefined_uexp: Exp.t = Undefined |> Exp.fresh;
+let undefined_menhir = () =>
+  alco_check_menhir("Undefined test (menhir)", undefined_str, undefined_uexp);
+
+let list_exp_str = "[1, 2, 3]";
+let list_exp_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term:
+    ListLit([
+      Int(1) |> Exp.fresh,
+      Int(2) |> Exp.fresh,
+      Int(3) |> Exp.fresh,
+    ]),
+  copied: false,
+};
+let list_exp_menhir = () =>
+  alco_check_menhir("List exp (menhir)", list_exp_str, list_exp_uexp);
+
+let invalid_str = "
+?\"x\"
+";
+let invalid_uexp: Exp.t = Invalid("x") |> Exp.fresh;
+let invalid_menhir = () =>
+  alco_check_menhir("Invalid test (menhir)", invalid_str, invalid_uexp);
+
+let ty_alias_str = "
+x
+";
+let ty_alias_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term:
+    TyAlias(
+      Var("x") |> TPat.fresh,
+      Int |> Typ.fresh,
+      Var("x") |> Exp.fresh,
     ),
+  copied: false,
+};
+let ty_alias_menhir = () =>
+  alco_check_menhir("Type alias test (menhir)", ty_alias_str, ty_alias_uexp);
+
+let list_concat_str = "[1, 2] @ [3, 4]";
+let list_concat_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term:
+    ListConcat(
+      ListLit([Int(1) |> Exp.fresh, Int(2) |> Exp.fresh]) |> Exp.fresh,
+      ListLit([Int(3) |> Exp.fresh, Int(4) |> Exp.fresh]) |> Exp.fresh,
+    ),
+  copied: false,
+};
+let list_concat_menhir = () =>
+  alco_check_menhir(
+    "List concat test (menhir)",
+    list_concat_str,
+    list_concat_uexp,
   );
 
+let unop_str = "-1";
+let unop_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term: UnOp(Int(Minus), Int(1) |> Exp.fresh),
+  copied: false,
+};
+let unop_menhir = () =>
+  alco_check_menhir("Unary operation test (menhir)", unop_str, unop_uexp);
+
+let seq_str = "1; 2";
+let seq_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term: Seq(Int(1) |> Exp.fresh, Int(2) |> Exp.fresh),
+  copied: false,
+};
+let seq_menhir = () =>
+  alco_check_menhir("Sequence test (menhir)", seq_str, seq_uexp);
+
+let fixf_str = "fix x -> 1<Int => Unknown Internal>";
+let fixf_uexp: Exp.t = {
+  ids: [id_at(0)],
+  term: FixF(Var("x") |> Pat.fresh, Int(1) |> Exp.fresh, None),
+  copied: false,
+};
+let fixf_menhir = () =>
+  alco_check_menhir("FixF test (menhir)", fixf_str, fixf_uexp);
+
 let elaboration_tests = [
-  test_case("Single integer", `Quick, single_integer),
-  test_case("Empty hole", `Quick, empty_hole),
-  test_case("Free variable", `Quick, free_var),
-  test_case("Let expression", `Quick, let_exp),
-  test_case("Inconsistent binary operation", `Quick, bin_op),
-  test_case("Consistent if statement", `Quick, consistent_if),
-  test_case("An unapplied function", `Quick, unapplied_function),
-  test_case("Application of function on free variable", `Quick, ap_fun),
-  test_case("Inconsistent case statement", `Quick, inconsistent_case),
-  test_case("Let expression for a function", `Quick, let_fun),
-  test_case(
-    "Function application with a deferred argument",
-    `Quick,
-    deferral,
-  ),
```

Comment:
It might be worth keeping the existing tests temporarily until we're fully confident in the menhir tests long term. Something like:
```reason
let tests = (
  "Menhir Elaboration",
  [
    test_case("Filter test (menhir)", `Quick, filter_menhir),
    test_case("Test failed (menhir)", `Quick, test_menhir),
    test_case("Built-in function (menhir)", `Quick, builtin_fun_menhir),
    test_case(
      "Dynamic error hole (menhir)",
      `Quick,
      dynamic_error_hole_menhir,
    ),
  ],
  "Elaboration",
  [
    test_case("Single integer", `Quick, single_integer),
    test_case("Empty hole", `Quick, empty_hole),
    test_case("Free variable", `Quick, free_var),
  ],
);
```

Or we can separate them into different modules. Let me know if you'd like me to make that change or if you'd want to.

Reply:
I'm mostly saying this because we're using the tests for the labeled tuple work to ensure we don't break anything.

Reply:
Sure, if you would like to make this change, that would be very helpful! Thanks!

Reply:
I did this and it's two modules inside the same file. It might make more sense to split the files up if it becomes unwieldy for you.